src/main/java/duke/Command.java

```diff
@@ -0,0 +1,148 @@
+            case "find":
+                response = taskList.searchKeyword(cmd[1]);
+                break;
+            //Event creation
+            case "todo":
+                response = createTodo(cmd, taskList);
+                break;
+            case "deadline":
+                response = createDeadline(cmd, taskList);
+                break;
+            case "event":
+                response = createEvent(cmd, taskList);
+                break;
+            case "delete":
+                response = ui.showTaskRemoved(taskList.removeTask(cmd[1]));
+                break;
+            case "bye":
+                isExiting = true;
+                break;
+            default:
+                //TODO: make it a exception
+                throw new DukeException("I'm not sure what that means :((");
+            }} catch (Exception e) {
+                response = e.getMessage();
+            } finally {
+            storage.saveList();
+            return response;
+        }
+    }
+
+
+    public boolean getExitStatus() {
+        return this.isExiting;
+    }
+
+//    public void validateTaskCreation(String[] cmd) {
+//        if (cmd.length != 2) {
+//    }
+
+    /**
+     * Creates a todo and adds it to TaskList
+     * @param cmd The parameters of the todo to be created
+     * @param taskList the TaskList which is active
+     * @throws DukeException
+     */
+    private String createTodo(String[] cmd, TaskList taskList) throws DukeException {
+        String response = "";
+        if (cmd.length == 2) {
+            Todo todo = new Todo(cmd[1]);
+            taskList.addTask(todo);
+            response = ui.showTaskAdded(todo);
+        } else {
+            throw new DukeException("The description of a todo cannot be empty.");
+        }
+        return response;
+    }
+
+    /**
+     * Creates a Deadline and adds it to TaskList
+     * @param cmd The parameters of the deadline to be created
+     * @param taskList the TaskList which is active
+     * @throws DukeException
+     */
+    private String createDeadline(String[] cmd, TaskList taskList) throws DukeException {
+        String response = "";
+        if (cmd.length == 2) {
+            String[] deadlineDetails = cmd[1].split(" /by ", 2);
+            if (deadlineDetails.length == 2) {
+                try {
+                    Deadline deadline = new Deadline(deadlineDetails[0], deadlineDetails[1]);
+                    taskList.addTask(deadline);
+                    response = ui.showTaskAdded(deadline);
+                } catch (DateTimeParseException e) {
+                    System.out.println(e.getMessage() + "\nPlease input deadline in yyyy-mm-dd");
+                }
+            } else {
+                throw new DukeException("Please indicate a time for the Deadline!");
+            }
+        } else {
+            throw new DukeException("The description of a Deadline date cannot be empty.");
+        }
+        return response;
```

Comment:
Perhaps this nested if statements could be refactored with guard clauses to make the happy path more apparent. Similarily with createEvent().
```suggestion
        String response = "";
        if (cmd.length < 2) {
            throw new DukeException("The description of a Deadline date cannot be empty.");
        }

        String[] deadlineDetails = cmd[1].split(" /by ", 2);

        if (deadlineDetails.length < 2) {
                throw new DukeException("Please indicate a time for the Deadline!");
        }

        try {
            Deadline deadline = new Deadline(deadlineDetails[0], deadlineDetails[1]);
            taskList.addTask(deadline);
            response = ui.showTaskAdded(deadline);
        } catch (DateTimeParseException e) {
            System.out.println(e.getMessage() + "\nPlease input deadline in yyyy-mm-dd");
        }
   
        return response;
```